google/resource_google_folder_iam_policy_test.go

```diff
@@ -0,0 +1,168 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	resourceManagerV2Beta1 "google.golang.org/api/cloudresourcemanager/v2beta1"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestAccGoogleFolderIamPolicy_basic(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+
+	folderDisplayName := "tf-test-" + acctest.RandString(10)
+	org := os.Getenv("GOOGLE_ORG")
+	parent := "organizations/" + org
+
+	policy := &resourceManagerV2Beta1.Policy{
+		Bindings: []*resourceManagerV2Beta1.Binding{
+			{
+				Role: "roles/viewer",
+				Members: []string{
+					"user:admin@hashicorptest.com",
+				},
+			},
+		},
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleFolderIamPolicyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGoogleFolderIamPolicy_basic(folderDisplayName, parent, policy),
+				Check:  testAccCheckGoogleFolderIamPolicy("google_folder_iam_policy.test", policy),
+			},
+		},
+	})
+}
+
+func TestAccGoogleFolderIamPolicy_update(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+
+	folderDisplayName := "tf-test-" + acctest.RandString(10)
+	org := os.Getenv("GOOGLE_ORG")
+	parent := "organizations/" + org
+
+	policy1 := &resourceManagerV2Beta1.Policy{
+		Bindings: []*resourceManagerV2Beta1.Binding{
+			{
+				Role: "roles/viewer",
+				Members: []string{
+					"user:admin@hashicorptest.com",
+				},
+			},
+		},
+	}
+	policy2 := &resourceManagerV2Beta1.Policy{
+		Bindings: []*resourceManagerV2Beta1.Binding{
+			{
+				Role: "roles/editor",
+				Members: []string{
+					"user:admin@hashicorptest.com",
```

Comment:
Does GCP validate that the domain exists? If not I'd prefer to generate a random email address/domain name here to avoid cases where a test doesn't clean up and leaves permissions open

alternatively if we can set an overriding policy rule that ensures nobody has access...

Reply:
It does check if a user exists.

Reply:
Bummer. For now hashicorptest.com works but at some point we should have a review of these sorts of tests